CraftingRevisions/Patches/GameManager_Awake.cs

```diff
@@ -1,19 +1,90 @@
-﻿using HarmonyLib;
+using HarmonyLib;
 using Il2Cpp;
-using MelonLoader;
+using System.Reflection;
+using UnityEngine;
 
 namespace CraftingRevisions.Patches
 {
-	[HarmonyPatch(typeof(GameManager), "Awake")]
-	internal class GameManager_Awake
+	// load the blueprints when the crafting panel is initialized
+	// (needed to be this late otherwise we have no instance of BlueprintManager)
+	[HarmonyPatch(typeof(Panel_Crafting), nameof(Panel_Crafting.Initialize))]
+	internal class Panel_Crafting_Initialize
 	{
 		private static void Postfix()
 		{
-			try { BlueprintMapper.MapBlueprints(); }
-			catch (System.Exception e)
+			// call TLD LoadAllUserBlueprints
+			InterfaceManager.m_Instance.m_BlueprintManager.LoadAllUserBlueprints();
+		}
+	}
+
+	// patched into postfix BlueprintManager.LoadAllUserBlueprints
+	// (BlueprintManager.RemoveUserBlueprints is not guaranteed to get called)
+	[HarmonyPatch(typeof(Il2CppTLD.Gear.BlueprintManager), nameof(Il2CppTLD.Gear.BlueprintManager.LoadAllUserBlueprints))]
+	internal class BlueprintManager_LoadAllUserBlueprints_Postfix
+	{
+		private static void Postfix(Il2CppTLD.Gear.BlueprintManager __instance)
+		{
+			// loop over the items
+			foreach (string jsonUserBlueprint in BlueprintManager.jsonUserBlueprints)
+			{
+
+				// load the blueprint into the game
+				bool loaded = __instance.LoadUserBlueprint(jsonUserBlueprint);
+
+				if (!loaded)
+				{
+					// validate the blueprint
+					BlueprintManager.ValidateJsonBlueprint(jsonUserBlueprint);
+				}
+			}
+		}
+	}
+
+	// These patches are to work around the fact that "UserBlueprintData.MakeRuntimeWwiseEvent" appears to be currently broken or unfinished
+	[HarmonyPatch(typeof(Il2CppTLD.Gear.UserBlueprintData), nameof(Il2CppTLD.Gear.UserBlueprintData.MakeRuntimeWwiseEvent), new Type[] { typeof(string) })]
+	internal class UserBlueprintData_MakeRuntimeWwiseEvent
+	{
+		private static void Prefix(ref bool __runOriginal)
+		{
+			__runOriginal = false;
+		}
+		private static void Postfix(string eventName, ref bool __runOriginal, ref Il2CppAK.Wwise.Event? __result)
+		{
+			__result = MakeAudioEvent(eventName);
+			__runOriginal = false;
+		}
```

Comment:
Two scenarios:

If the game throws an exception whenever the original is run:

* Move the code into the prefix and delete the postfix.

If the game gracefully returns null whenever the original is run:

* Delete the prefix and only make the event whenever Hinterland doesn't.
* If it *always* returns null, add a log statement to remind us in the future when this gets implemented.